addons/webhook/certificates.go

```diff
@@ -0,0 +1,125 @@
+	rsrcs "knative.dev/pkg/webhook/certificates/resources"
+	"path"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"time"
+)
+
+const (
+	ServerKey = "server-key.pem"
+	// ServerCert is the name of the key associated with the secret's public key.
+	ServerCert = "server-cert.pem"
+	// CACert is the name of the key associated with the certificate of the CA for
+	// the keypair.
+	CACert = "ca-cert.pem"
+)
+
+func SetupWebhooks(mgr *manager.Manager, namespace string, vWhCfgNames, mWhCfgNames []string) error {
+
+	//create the secret and write the certificate files based on the existing webhook server config
+	wh := (*mgr).GetWebhookServer()
+	secret, err := UpdateTLSSecret("addons-webhook-tls", wh, namespace)
+	if err != nil {
+		return err
+	}
+
+	//update the webhooks with the new CA from the secret created
+	k8sConfig, err := ctrl.GetConfig()
+	if err != nil {
+		return err
+	}
+	client, err := kubernetes.NewForConfig(k8sConfig)
+	if err != nil {
+		return err
+	}
+
+	for _, name := range vWhCfgNames {
+		err := UpdateValidatingWebhook(client, name, secret)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, name := range mWhCfgNames {
+		err := UpdateMutatingWebhook(client, name, secret)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func UpdateValidatingWebhook(client kubernetes.Interface, name string, secret *corev1.Secret) error {
+
+	vWhCfg, err := client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	for idx, webhook := range vWhCfg.Webhooks {
+		webhook.ClientConfig.CABundle = secret.Data[CACert]
+		vWhCfg.Webhooks[idx] = webhook
+	}
+	if _, err := client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Update(context.TODO(), vWhCfg, metav1.UpdateOptions{}); err != nil {
+		return fmt.Errorf("error updating addons CA cert of ValidatingWebhookConfiguration %s: %v", name, err)
+	}
+
+	return nil
+}
+
+func UpdateMutatingWebhook(client kubernetes.Interface, name string, secret *corev1.Secret) error {
+	mWhCfg, err := client.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	for idx, webhook := range mWhCfg.Webhooks {
+		webhook.ClientConfig.CABundle = secret.Data[CACert]
+		mWhCfg.Webhooks[idx] = webhook
+	}
+	if _, err := client.AdmissionregistrationV1().MutatingWebhookConfigurations().Update(context.TODO(), mWhCfg, metav1.UpdateOptions{}); err != nil {
+		return fmt.Errorf("error updating addons CA cert of MutatingWebhookConfiguration %s: %v", name, err)
+	}
+
+	return nil
+}
+
+//func UpdateTLSSecret(ctx context.Context, name, namespace string, maxAge time.Duration, certDir, tlsCertName, tlsKeyName string) (*corev1.Secret, error) {
+func UpdateTLSSecret(name string, wh *webhook.Server, namespace string) (*corev1.Secret, error) {
```

Comment:
It would be great to be able to use the create secret from https://github.com/knative/pkg/blob/3c00bb0157b9242359df6868e720861b6807773f/webhook/certificates/resources/secret.go#L45, but that one has a timelimit of one week hard coded, also we would need to create exctra functions to write the cert to the correct files. Might as well do it here below and be done.

Reply:
Seems like it was due to [recent security change](https://github.com/knative/pkg/pull/1998) where expiry is 7 days and rotation is 1 day before expiry. Seems like something we should adopt as well.

Reply:
I think is best to use the one provided by knative then. I'll make the changes.

Reply:
replaced the with the the mentioned "create secret" from knative/pkg